### PR TITLE
[Rules] Add Keep Level on Death

### PR DESCRIPTION
**Pull request by trentdm**
I still want to do some testing around this, but am putting this out as a draft PR to get feedback on whether this is something that would be desired or not.

### CODE REVIEW

File: utils/sql/db_update_manifest.txt [hunk 1]
```diff
@@ -443,6 +443,7 @@
 9187|2022_07_09_task_zone_version_matching.sql|SHOW COLUMNS FROM `task_activities` LIKE 'zone_version'|empty|
 9188|2022_07_14_zone_expansion_revert.sql|SHOW COLUMNS FROM `zone` LIKE 'expansion'|empty|
 9189|2022_07_10_character_task_rewarded.sql|SHOW COLUMNS FROM `character_tasks` LIKE 'was_rewarded'|empty|
+9190|2022_07_18_rule_death_keep_level.sql|SELECT * FROM `rule_values` WHERE `rule_name` LIKE '%Character:DeathKeepLevel%'|empty|
```
Comment (mackal) [inline utils/sql/db_update_manifest.txt:446]: Rules are automatically inserted, this is unneeded, should be removed

**Review by mackal — CHANGES_REQUESTED** [review]

**Comment by nytmyr** [comment]
Way I look at it is, someone somewhere will use it and anything custom hid behind a rule just adds to the simple customization of a server.

**Comment by Akkadius** [comment]
If you are ready for review, this is still in draft

File: utils/sql/db_update_manifest.txt (hunk 1 above)
Comment (trentdm) [reply utils/sql/db_update_manifest.txt]: @mackal , I also removed the migration file, https://github.com/EQEmu/Server/pull/2319/commits/de93b545fab621623aa3d97aba7f98481404ecd9, but should I have kept that in?

I'm trying to test locally but not seeing the rule added in the database, but very well am doing something else wrong in getting this set up.

**Comment by trentdm** [comment]
> If you are ready for review, this is still in draft

Sorry for the long delay, I've been meaning to test locally first and haven't had time. I'm planning to work on it this weekend.

File: zone/attack.cpp [hunk 2]
```diff
@@ -1843,6 +1843,15 @@ bool Client::Death(Mob* killerMob, int64 damage, uint16 spell, EQ::skills::Skill
 		exploss *= zone->level_exp_mod[GetLevel()].ExpMod;
 	}
 
+	if (exploss > 0 && RuleB(Character, DeathKeepLevel)) {
+		int32 totalExp = GetEXP();
```
Comment (Akkadius) [inline zone/attack.cpp:1847]: `snake_case` eg `total_exp` for locally scoped variables

Same for below variables

**Review by Akkadius — APPROVED** [review]

**Pull request merged**